### PR TITLE
Nds 811 remove description from nav

### PR DESCRIPTION
**Pull request by coostenbrug**
Intent: 
 - review to merge
This PR:
 - removes description support from NavBar
 - cleans up styling on NavBar related components

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Changed
 ### Deprecated
 ### Removed
+- Description prop in the [`NavBar`](http://localhost:8080/?selectedKind=NavBar&selectedStory=NavBar) component has been removed
```
Comment (matthewlyle) [inline CHANGELOG.md]: I think this should either be referencing our live documentation site or the live storybook. Probably the docs site?

File: components/src/NavBar/SubMenu.js [hunk 2]
```diff
@@ -76,24 +76,22 @@ BaseSubMenu.defaultProps = {
...
-    display: "flex",
-    maxWidth: "20em",
-    flexDirection: "column",
-    fontSize: theme.fontSizes.small,
     backgroundColor: subMenuStyles.backgroundColor,
     borderRadius: theme.radii.medium,
     borderTop: `1px solid ${subMenuStyles.borderColor}`,
     borderBottom: `1px solid ${subMenuStyles.borderColor}`,
-    boxShadow: theme.boxShadows,
+    boxShadow: theme.boxShadows.small,
```
Comment (matthewlyle) [inline components/src/NavBar/SubMenu.js]: Nikola just merged a branch this morning that changes `boxShadows` to `shadows`.

Comment (coostenbrug) [reply components/src/NavBar/SubMenu.js]: rebased and made the fix

File: CHANGELOG.md (hunk 1 above)
Comment (coostenbrug) [reply CHANGELOG.md]: changed to docs link

**Review by matthewlyle — APPROVED** [review]

**Pull request merged**